README.md
## Angular Electron Admin

Bootstrap and package your project with Angular 12 and Electron 13 (Typescript + SCSS + HMR) for creating Desktop applications.

|Project|Version|
|---|---|
|Angular|12.x|
|Electron|13.x|
|Electron Builder|22.x|
|Bootstrap|4.x|

## Getting Started

- Fork project for your
- Clone the project from GitHub
```
git clone https://github.com/<GitHub_UserID>/angular-electron-admin.git
```

- Install the node dependencies
```
cd angular-electron-admin
yarn install
```

- Start application
```
yarn run serve
```

- Release the application
```
yarn run release
```


CHANGELOG.md
## 1.3.0-SNAPSHOT (2021-11-23)

* [Bug] Fix the blank screen caused by sub-page loading ([4265d8e](https://github.com/qianmoQ/angular-electron-admin/commit/4265d8e))
* [Bug] Fixed file not found in package ([2ee9253](https://github.com/qianmoQ/angular-electron-admin/commit/2ee9253))
* [Bug] Fixed some of the sample button style squeeze effects ([cf354d9](https://github.com/qianmoQ/angular-electron-admin/commit/cf354d9))
* [CI] Add auto changelog cli ([3c3609a](https://github.com/qianmoQ/angular-electron-admin/commit/3c3609a))
* [CI] Add issues template ([ca8e1a1](https://github.com/qianmoQ/angular-electron-admin/commit/ca8e1a1))
* [CI] Add publish ci script ([a90fad2](https://github.com/qianmoQ/angular-electron-admin/commit/a90fad2))
* [CI] Add PULL_REQUEST_TEMPLATE ([5d4e4c3](https://github.com/qianmoQ/angular-electron-admin/commit/5d4e4c3))
* [Component] Add Accordion component ([b777c97](https://github.com/qianmoQ/angular-electron-admin/commit/b777c97))
* [Component] Add Alert component ([c372b02](https://github.com/qianmoQ/angular-electron-admin/commit/c372b02))
* [Component] Add Collapse component ([b3fff83](https://github.com/qianmoQ/angular-electron-admin/commit/b3fff83))
* [Component] Add component for carousel ([840ea41](https://github.com/qianmoQ/angular-electron-admin/commit/840ea41))
* [Component] Add component for clipboard ([2457f12](https://github.com/qianmoQ/angular-electron-admin/commit/2457f12))
* [Component] Add component for datepicker ([c2ea384](https://github.com/qianmoQ/angular-electron-admin/commit/c2ea384))
* [Component] Add component for keyboard ([d455fb6](https://github.com/qianmoQ/angular-electron-admin/commit/d455fb6))
* [Component] Add component for Progressbar ([d661f3e](https://github.com/qianmoQ/angular-electron-admin/commit/d661f3e))
* [Component] Add component for scrollbar ([ec9a395](https://github.com/qianmoQ/angular-electron-admin/commit/ec9a395))
* [Component] Add component for tooltips ([f202e70](https://github.com/qianmoQ/angular-electron-admin/commit/f202e70))
* [Component] Add Dropdown component ([ab8070d](https://github.com/qianmoQ/angular-electron-admin/commit/ab8070d))
* [Component] Add Popover component ([b74626c](https://github.com/qianmoQ/angular-electron-admin/commit/b74626c))
* [Component] Add Rating component ([064e712](https://github.com/qianmoQ/angular-electron-admin/commit/064e712))
* [Component] Add Sortable component ([4a08047](https://github.com/qianmoQ/angular-electron-admin/commit/4a08047))
* [Component] Add Tabs component ([08b4176](https://github.com/qianmoQ/angular-electron-admin/commit/08b4176))
* [Component] Add Timepicker component ([08b85e0](https://github.com/qianmoQ/angular-electron-admin/commit/08b85e0))
* [Component] Add Toasty component ([9823f13](https://github.com/qianmoQ/angular-electron-admin/commit/9823f13))
* [Component] Add Typeahead component ([aa5a5f4](https://github.com/qianmoQ/angular-electron-admin/commit/aa5a5f4))
* [Directive] Add contribution chart ([3326ed9](https://github.com/qianmoQ/angular-electron-admin/commit/3326ed9))
* [Directive] Add screenshot ([12c5d65](https://github.com/qianmoQ/angular-electron-admin/commit/12c5d65))
* [Directive] Custom macros use the version specified in package.json ([8b37cb4](https://github.com/qianmoQ/angular-electron-admin/commit/8b37cb4))
* [Error] Add Forbidden(403) page ([c32006a](https://github.com/qianmoQ/angular-electron-admin/commit/c32006a))
* [Error] Add NotFound(404) page ([bee29dc](https://github.com/qianmoQ/angular-electron-admin/commit/bee29dc))
* [Feature] Rebuild the Form element ([f34a527](https://github.com/qianmoQ/angular-electron-admin/commit/f34a527))
* [Icon] Add font-awesome ([d91c00b](https://github.com/qianmoQ/angular-electron-admin/commit/d91c00b))
* [Icon] Add material-design-iconic-font ([bf537e3](https://github.com/qianmoQ/angular-electron-admin/commit/bf537e3))
* [Init] Add CODE_OF_CONDUCT and LICENSE ([26d136e](https://github.com/qianmoQ/angular-electron-admin/commit/26d136e))
* [Init] Add yarn.lock ([260b7ad](https://github.com/qianmoQ/angular-electron-admin/commit/260b7ad))
* [Init] Init aea(angular-electron-admin) project ([f3a42b1](https://github.com/qianmoQ/angular-electron-admin/commit/f3a42b1))
* [Layout] Add a left menu scroll bar ([7384602](https://github.com/qianmoQ/angular-electron-admin/commit/7384602))
* [Layout] Add basic layout ([56e2318](https://github.com/qianmoQ/angular-electron-admin/commit/56e2318))
* [Layout] Add left navigation layout ([4a90c8b](https://github.com/qianmoQ/angular-electron-admin/commit/4a90c8b))
* [Layout] Add small screen menu control ([74d93ce](https://github.com/qianmoQ/angular-electron-admin/commit/74d93ce))
* [Layout] Add theme ([8f33cd0](https://github.com/qianmoQ/angular-electron-admin/commit/8f33cd0))
* [Optimize] Fixed SCSS file compilation warning prompt ([6badda5](https://github.com/qianmoQ/angular-electron-admin/commit/6badda5))
* [Optimize] Optimize README ([bcf6faa](https://github.com/qianmoQ/angular-electron-admin/commit/bcf6faa))





src/renderer/app/layout/layout.routing.ts
import { RouterModule, Routes } from '@angular/router';
import { LayoutComponent } from './layout.component';
import { ForbiddenComponent } from '@renderer/app/pages/error/forbidden/forbidden.component';
import { NotfoundComponent } from '@renderer/app/pages/error/notfound/notfound.component';

const LAYOUT_ROUTES: Routes = [
  {
    path: '', component: LayoutComponent, children: [
      {
        path: '',
        redirectTo: '/home',
        pathMatch: 'full'
      },
      {
        path: 'home',
        loadChildren: () => import('../pages/home/home.module').then(m => m.HomeModule)
      },
      {
        path: 'icon',
        children: [
          {
            path: 'zmdi',
            loadChildren: () => import('../pages/icon/zmdi/zmdi.module').then(m => m.IconsZmdiModule)
          },
          {
            path: 'fa',
            loadChildren: () => import('../pages/icon/fa/fa.module').then(m => m.IconsFaModule)
          }
        ]
      },
      {
        path: 'component',
        children: [
          {
            path: 'clipboard',
            loadChildren: () => import('../pages/component/clipboard/clipboard.module').then(m => m.ComponentClipboardModule)
          },
          {
            path: 'tooltips',
            loadChildren: () => import('../pages/component/tooltips/tooltips.module').then(m => m.TooltipsComponentModule)
          },
          {
            path: 'scrollbar',
            loadChildren: () => import('../pages/component/scrollbar/scrollbar.module').then(m => m.ScrollbarModule)
          },
          {
            path: 'carousel',
            loadChildren: () => import('../pages/component/carousel/carousel.module').then(m => m.CarouselComponentModule)
          },
          {
            path: 'progressbar',
            loadChildren: () => import('../pages/component/progressbar/progressbar.module').then(m => m.ProgressbarComponentModule)
          },
          {
            path: 'keyboard',
            loadChildren: () => import('../pages/component/keyboard/keyboard.module').then(m => m.KeyboardModule)
          },
          {
            path: 'sortable',
            loadChildren: () => import('../pages/component/sortable/sortable.module').then(m => m.SortableComponentModule)
          },
          {
            path: 'alert',
            loadChildren: () => import('../pages/component/alert/alert.module').then(m => m.AlertsComponentModule)
          },
          {
            path: 'tabs',
            loadChildren: () => import('../pages/component/tabs/tabs.module').then(m => m.TabsComponentModule)
          },
          {
            path: 'toasty',
            loadChildren: () => import('../pages/component/toasty/toasty.module').then(m => m.ToastyComponentModule)
          },
          {
            path: 'typeahead',
            loadChildren: () => import('../pages/component/typeahead/typeahead.module').then(m => m.TypeaheadComponentModule)
          },
          {
            path: 'accordion',
            loadChildren: () => import('../pages/component/accordion/accordion.module').then(m => m.AccordionComponentModule)
          },
          {
            path: 'collapse',
            loadChildren: () => import('../pages/component/collapse/collapse.module').then(m => m.CollapseComponentModule)
          },
          {
            path: 'popover',
            loadChildren: () => import('../pages/component/popover/popover.module').then(m => m.PopoverComponentModule)
          },
          {
            path: 'modals',
            loadChildren: () => import('../pages/component/modals/modals.module').then(m => m.ModalsModule)
          }
        ]
      },
      {
        path: 'form',
        children: [
          {
            path: 'datepicker',
            loadChildren: () => import('../pages/form/datepicker/datepicker.module').then(m => m.DatepickerModule)
          },
          {
            path: 'timepicker',
            loadChildren: () => import('../pages/form/timepicker/timepicker.module').then(m => m.TimepickerComponentModule)
          },
          {
            path: 'rating',
            loadChildren: () => import('../pages/form/rating/rating.module').then(m => m.RatingComponentModule)
          },
          {
            path: 'dropdown',
            loadChildren: () => import('../pages/form/dropdown/dropdown.module').then(m => m.DropdownComponentModule)
          },
          {
            path: 'ngx-tree',
            loadChildren: () => import('../pages/form/tree/ngx-tree/ngx-tree.module').then(m => m.NgxTreeModule)
          }
        ]
      },
      {
        path: 'directive',
        children: [
          {
            path: 'contribution',
            loadChildren: () => import('../pages/directive/contribution/contribution.module').then(m => m.DirectiveContributionModule)
          },
          {
            path: 'screenshot',
            loadChildren: () => import('../pages/directive/screenshot/screenshot.module').then(m => m.DirectiveScreenshotModule)
          }
        ]
      },
      {
        path: 'error',
        children: [
          {
            path: '403',
            component: ForbiddenComponent
          },
          {
            path: '404',
            component: NotfoundComponent
          }
        ]
      },
      {
        path: 'other',
        children: [
          {
            path: 'i18n',
            loadChildren: () => import('../pages/other/i18n/i18n.module').then(m => m.I18nModule)
          }
        ]
      },
      {
        path: 'editor',
        children: [
          {
            path: 'codemirror',
            loadChildren: () => import('../pages/editor/codemirror/codemirror.module').then(m => m.CodemirrorComponentModule)
          }
        ]
      }
    ]
  }
];
// eslint-disable-next-line @typescript-eslint/naming-convention
export const LayoutRouting = RouterModule.forChild(LAYOUT_ROUTES);
